torch/_dynamo/variables/constant.py

```diff
@@ -37,18 +36,29 @@
 class ConstantVariable(VariableTracker):
     @staticmethod
     def create(value, **kwargs) -> VariableTracker:
+        """
+        Create a `ConstantVariable` based on the given value.
+
+        NOTE: the caller must install the proper guards if needed; most often
+        the guard will be `CONSTANT_MATCH`.
+        """
         source = kwargs.get("source", None)
 
         # Routing for supported collection literals.
+        #
+        # TODO for the items, we should probably use
+        # `LazyVariableTracker.create` or `VariableTracker.build` here rather
+        # than `ConstantVariable.create`, which could be too restrictive, e.g.,
+        # say the item is a function that gets loaded and called elsewhere
+        # later (`ConstantVariable` doesn't implement `call_function`). This
+        # interacts in a subtle way with `VariableBuilder._wrap`.
         if isinstance(value, (set, frozenset)):
             items = [ConstantVariable.create(x) for x in value]
             return variables.SetVariable(items, **kwargs)
         elif isinstance(value, (list, tuple)):
             items = []
             for i, x in enumerate(value):
                 item_source = GetItemSource(source, i) if source else None
-                if item_source:
-                    install_guard(item_source.make_guard(GuardBuilder.CONSTANT_MATCH))
                 items.append(
```

Comment:
I can't even figure out a test for this path, because it never fires for tests in `test/dynamo`...

Reply:
If you think it is dead code maybe try removing it.